### ISSUE TITLE
Improve Importers Progress Management

### ISSUE THREAD

**Issue by mwestphal** [f3d-app/f3d#22]

**Comment by mwestphal** [comment]
Good first issue for a VTK developer

**Comment by mwestphal** [comment]
glTF does not have progress either

**Comment by karchit0509** [comment]
Hi, I was wondering if I could work on this?

**Comment by Meakk** [comment]
Sure. Go ahead. Please free to reach out on discord if you need help.

**Comment by mwestphal** [comment]
HI @karchit0509 

Do you need any help moving forward ?

**Comment by karchit0509** [comment]
Hi Mathieu,

Thanks so much for reaching out. At the moment we are good but we plan to
do a large chunk of work this week so I will keep you updated.

Thanks again,
Archit

**Comment by mwestphal** [comment]
HI @karchit0509

Do you need any help moving forward ?

**Comment by karchit0509** [comment]
Hi Mathieu,

Thanks for reaching out. I have been working on investigating all the
readers and importers in use to see which ones have progress reporting and
implementing the progress reporting mechanism starting with OBJ and gITF. I
am wondering where in the folder I should add the code for each of these
(i.e should I create a new file and import the progress checking into the
original "reader_OBJ.h" file and if so where in the directory should I add
this file or should I just try to implement it directly in the
"reader_OBJ.h file").

Thanks for your help,
Archit

**Comment by mwestphal** [comment]
> the progress reporting mechanism starting with OBJ and gITF. I am wondering where in the folder I should add the code

This should be done and contributed in VTK -> gitlab.kitware.com/vtk/vtk

**Comment by karchit0509** [comment]
Hi Mathieu,

I hope you are doing well. I've compiled a table summarizing the readers in
use, their file types, descriptions, progress reporting status, and
dependencies based on some investigating that I did. I don't know if these
are perfectly accurate but I did my best.

Best regards,
Archit Kumar

**Comment by mwestphal** [comment]
Great! Could you share it here ?

**Comment by mwestphal** [comment]
Hi @karchit0509 , any news on this ?

**Comment by Hy-LeTuan** [comment]
Hey @mwestphal, could I be assigned this issue as discussed on Discord?

**Comment by mwestphal** [comment]
https://docs.google.com/spreadsheets/d/1lzSzjC2alSz_Z7NWNFEgr-OY9il9ShAQOb9TOXreoyI/edit?gid=426098779#gid=426098779

**Comment by mwestphal** [comment]
Still working on this @Hy-LeTuan ? :)

**Comment by Hy-LeTuan** [comment]
Yup I'm trying to figure out the VRML parser, the yacc code is just giving me a really hard time 😅

**Comment by mwestphal** [comment]
Unassingning after no feedback from @Hy-LeTuan

**Comment by Hy-LeTuan** [comment]
Sorry for not completing the whole issue, I currently don't have the free time to work on it right now.

I have updated the document: https://docs.google.com/spreadsheets/d/1lzSzjC2alSz_Z7NWNFEgr-OY9il9ShAQOb9TOXreoyI/edit?gid=426098779#gid=426098779 to include the progresses I've made, suggested fixes and other details on the issue for anyone who works on this after!

**Comment by mwestphal** [comment]
@Hy-LeTuan that's ok! thanks for sharing and for the accomplished work so far :)

**Comment by thelegendaryposeidon** [comment]
This still open and free to be assigned? I'd like to take a shot at this. I'll start with devising my execution pathway upon assignment. I'll probably need some amount of course correction though, as this is my first attempt at open source.

**Comment by mwestphal** [comment]
> This still open and free to be assigned? I'd like to take a shot at this. I'll start with devising my execution pathway upon assignment. I'll probably need some amount of course correction though, as this is my first attempt at open source.

Of course, no worries

**Comment by mwestphal** [comment]
Need any help moving forward  @thelegendaryposeidon ?

**Comment by thelegendaryposeidon** [comment]
> Need any help moving forward [@thelegendaryposeidon](https://github.com/thelegendaryposeidon) ?

Sorry for the delay, I got bogged down by university stuff. I'm still actively making progress on the issue nevertheless. I'll be sure to reach out when I do encounter any hurdles.

**Comment by mwestphal** [comment]
Need any help moving forward @thelegendaryposeidon ?

**Comment by thelegendaryposeidon** [comment]
> Need any help moving forward @thelegendaryposeidon ?

Almost there, I'll get atleast a partial review request out over the weekend.

**Comment by mwestphal** [comment]
Need any help moving forward @thelegendaryposeidon ?

**Comment by thelegendaryposeidon** [comment]
> Need any help moving forward @thelegendaryposeidon ?

Still working, I'm really regretful for the delay. had my semester exams come up and university got in the way. Vacations are on the horizon though I'd greatly appreciate if you'd let me hold onto this issue up until then.

**Comment by mwestphal** [comment]
Of course, no worries!

**Comment by mwestphal** [comment]
Need any help moving forward @thelegendaryposeidon ?

**Comment by thelegendaryposeidon** [comment]
> Need any help moving forward [@thelegendaryposeidon](https://github.com/thelegendaryposeidon) ?

Yep, I've been reaching out to past contributors whenever I feel like I'm getting bogged down. Making steady progress so far.

**Comment by mwestphal** [comment]
Well, let me know if you need help!
